From ab03be1a4833eeb3c095c0506fbc80983a55113e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Milan=20P=2E=20Stani=C4=87?= <mps@arvanta.net>
Date: Sat, 1 Nov 2025 09:21:47 +0000
Subject: [PATCH] src/x86/freq/freq.c: pthread_attr_setaffinity_np is not POSIX
 defined

pthread_attr_setaffinity_np is glibcism not in POSIX standard
in musl libc pthread_setaffinity_np function is used
---
 src/x86/freq/freq.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/x86/freq/freq.c b/src/x86/freq/freq.c
index 308b8af..3db1adb 100644
--- a/src/x86/freq/freq.c
+++ b/src/x86/freq/freq.c
@@ -158,8 +158,8 @@ int32_t measure_frequency(struct cpuInfo* cpu, int32_t *max_freq_pp_vec) {
     // on the correct core.
     CPU_ZERO(&cpus);
     CPU_SET(i, &cpus);
-    if ((ret = pthread_attr_setaffinity_np(&attr, sizeof(cpu_set_t), &cpus)) != 0) {
-      printErr("pthread_attr_setaffinity_np: %s", strerror(ret));
+    if ((ret = pthread_setaffinity_np(&attr, sizeof(cpu_set_t), &cpus)) != 0) {
+      printErr("pthread_setaffinity_np: %s", strerror(ret));
       return -1;
     }
 
-- 
2.51.2

